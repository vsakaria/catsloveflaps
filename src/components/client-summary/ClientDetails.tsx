import React from "react";
import { IClient } from "../IClient";

const ClientDetails: React.FC<IClient> = ({
  id,
  name,
  status,
  adminType,
  passwordExpiryPeriod,
  activeFrom
}) => {
  return (
    <section className="clientDetailContainer">
      <h4 className="title">Client details</h4>
      <table>
        <tbody>
          <tr>
            <td>Client id</td>
            <td className="bold">{id.toString()}</td>
          </tr>
          <tr>
            <td>Client name</td>
            <td className="bold">{name}</td>
          </tr>
          <tr>
            <td>Client status</td>
            <td className="bold">{status}</td>
          </tr>
          <tr>
            <td>Administration type</td>
            <td className="bold">{adminType}</td>
          </tr>
          <tr>
            <td>Password expiry period</td>
            <td className="bold">{`${passwordExpiryPeriod} ${
              passwordExpiryPeriod > 1 ? "days" : "day"
            }`}</td>
          </tr>
          <tr>
            <td>Active form</td>
            <td className="bold">{activeFrom}</td>
          </tr>
        </tbody>
      </table>
    </section>
  );
};

export default ClientDetails;
