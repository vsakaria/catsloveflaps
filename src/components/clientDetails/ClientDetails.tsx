import React from "react";
import { withErrorBoundary } from "../../utils/components/ErrorBoundaryHoC";
import { Client } from "../IClient";
import styles from "./styles/ClientDetails.module.css";

export const ClientDetails: React.FC<Client> = ({
  id,
  name,
  status,
  adminType,
  passwordExpiryPeriod,
  activeFrom
}) => {
  return (
    <section className={styles.clientDetailContainer}>
      <h4 className="primary-color">Client details</h4>
      <table className={styles.clientDetailTable}>
        <tbody>
          <tr data-testid="client-id">
            <td>Client id</td>
            <td>{id}</td>
          </tr>
          <tr data-testid="client-name">
            <td>Client name</td>
            <td>{name}</td>
          </tr>
          <tr>
            <td>Client status</td>
            <td>{status}</td>
          </tr>
          <tr>
            <td>Administration type</td>
            <td>{adminType}</td>
          </tr>
          <tr>
            <td>Password expiry period</td>
            <td>{`${passwordExpiryPeriod} ${
              passwordExpiryPeriod > 1 ? "days" : "day"
            }`}</td>
          </tr>
          <tr>
            <td>Active form</td>
            <td>{activeFrom}</td>
          </tr>
        </tbody>
      </table>
    </section>
  );
};

export default withErrorBoundary(ClientDetails);
