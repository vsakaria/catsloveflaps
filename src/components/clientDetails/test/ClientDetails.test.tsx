import React from "react";
import { ClientDetails } from "../ClientDetails";
import { render, cleanup } from "../../../utils/testing/testingUtlis";

afterEach(cleanup);

describe("ClientSummary", () => {
  const props = {
    id: 49084369,
    name: "AutoCBS1",
    status: "Active",
    adminType: "Single Admin",
    passwordExpiryPeriod: 1,
    activeFrom: "24/05/2019"
  };

  it("match the snapshot", () => {
    const { asFragment } = render(<ClientDetails {...props} />);
    expect(asFragment()).toMatchSnapshot("Normal props with password expiry 1");
  });

  it("renders 90 days text", () => {
    props.passwordExpiryPeriod = 90;
    const { asFragment } = render(<ClientDetails {...props} />);
    expect(asFragment()).toMatchSnapshot(
      "Normal props with password expiry 90"
    );
  });

  it("inserts text in h1", () => {
    const { getByText } = render(<ClientDetails {...props} />);
    expect(getByText("Client details")).toHaveClass("primary-color");
  });
});
