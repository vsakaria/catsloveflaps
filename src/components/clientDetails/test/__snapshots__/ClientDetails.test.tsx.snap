// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientSummary match the snapshot: Normal props with password expiry 1 1`] = `
<DocumentFragment>
  <section
    class="clientDetailContainer"
  >
    <h4
      class="primary-color"
    >
      Client details
    </h4>
    <table
      class="clientDetailTable"
    >
      <tbody>
        <tr
          data-testid="client-id"
        >
          <td>
            Client id
          </td>
          <td>
            49084369
          </td>
        </tr>
        <tr
          data-testid="client-name"
        >
          <td>
            Client name
          </td>
          <td>
            AutoCBS1
          </td>
        </tr>
        <tr>
          <td>
            Client status
          </td>
          <td>
            Active
          </td>
        </tr>
        <tr>
          <td>
            Administration type
          </td>
          <td>
            Single Admin
          </td>
        </tr>
        <tr>
          <td>
            Password expiry period
          </td>
          <td>
            1 day
          </td>
        </tr>
        <tr>
          <td>
            Active form
          </td>
          <td>
            24/05/2019
          </td>
        </tr>
      </tbody>
    </table>
  </section>
</DocumentFragment>
`;

exports[`ClientSummary renders 90 days text: Normal props with password expiry 90 1`] = `
<DocumentFragment>
  <section
    class="clientDetailContainer"
  >
    <h4
      class="primary-color"
    >
      Client details
    </h4>
    <table
      class="clientDetailTable"
    >
      <tbody>
        <tr
          data-testid="client-id"
        >
          <td>
            Client id
          </td>
          <td>
            49084369
          </td>
        </tr>
        <tr
          data-testid="client-name"
        >
          <td>
            Client name
          </td>
          <td>
            AutoCBS1
          </td>
        </tr>
        <tr>
          <td>
            Client status
          </td>
          <td>
            Active
          </td>
        </tr>
        <tr>
          <td>
            Administration type
          </td>
          <td>
            Single Admin
          </td>
        </tr>
        <tr>
          <td>
            Password expiry period
          </td>
          <td>
            90 days
          </td>
        </tr>
        <tr>
          <td>
            Active form
          </td>
          <td>
            24/05/2019
          </td>
        </tr>
      </tbody>
    </table>
  </section>
</DocumentFragment>
`;
